sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -191,12 +197,21 @@ case class CreateViewCommand(
       // 2) The temp functions are represented by multiple classes. Most are inaccessible from this
       // package (e.g., HiveGenericUDF).
       child.collect {
-        // Disallow creating permanent views based on temporary views.
+        // Permanent views will be created as temporary view if based on temporary view.
         case UnresolvedRelation(AsTableIdentifier(ident))
             if sparkSession.sessionState.catalog.isTemporaryTable(ident) =>
-          // temporary views are only stored in the session catalog
-          throw new AnalysisException(s"Not allowed to create a permanent view $name by " +
-            s"referencing a temporary view $ident")
+          // Temporary views are only stored in the session catalog
+          if (sparkSession.sqlContext.conf.usePostgreSQLDialect) {
```

Comment:
I have handled pgSQLDialect case here. Since class `CreateViewCommand` of sql/core is not accessible from `PostgreSQLDialect` sql/catalyst because sql/core is dependent on sql/catalyst.